lib/cassandra-schema/migrator.rb

```diff
@@ -2,19 +2,30 @@
 
 module CassandraSchema
   class Migrator
-    attr_reader :connection, :current_version
+    attr_reader :connection, :current_version, :options
 
-    def initialize(connection:, migrations:, logger: Logger.new(STDOUT))
+    DEFAULT_OPTIONS = {
+      lock:         true,
+      lock_timeout: 30,
+    }
+
+    def initialize(connection:, migrations:, logger: Logger.new(STDOUT), options: {})
       @connection = connection
       @logger     = logger
       @migrations = migrations
+      @options    = DEFAULT_OPTIONS.merge(options)
 
       generate_migrator_schema!
-
-      @current_version = get_current_version || init_versioning
     end
 
     def migrate(target = nil)
+      if @options.fetch(:lock) && !lock_schema
+        @logger.info "Can't run migrations. Schema is locked."
+        return
```

Comment:
Should we consider a retry logic here, that keeps trying until a total timeout?

Reply:
Interesting idea, and It shouldn't be hard to implements. What do you think if I tackle it on a different issue/PR?

Reply:
I created a new issue to track progress for this idea #4